ipatests/test_ipaserver/test_install/test_installutils.py

```diff
@@ -61,15 +62,21 @@ def gpgkey(request, tempdir):
     )
 
     def fin():
-        if orig_gnupghome is not None:
-            os.environ['GNUPGHOME'] = orig_gnupghome
-        else:
-            os.environ.pop('GNUPGHOME', None)
         subprocess.run(
             [paths.GPG_CONF, '--kill', 'all'],
             check=True,
             env=env,
         )
+        path = os.path.join(os.environ['GNUPGHOME'], 'S.gpg-agent.ssh')
+        try:
+            while os.stat(path):
+                sleep(1)
```

Comment:
may result in an infinite loop.

Reply:
Yep. Yet, we need to wait here. Any other proposal would be great. Using GPG client might not work here...